engine/class_modules/sc_death_knight.cpp

```diff
@@ -5297,6 +5311,11 @@ struct obliterate_strike_t : public death_knight_melee_attack_t
     {
       p() -> buffs.icy_citadel_builder -> trigger();
     }
+
+    if ( p()->buffs.killing_machine->up() )
+    {
+      school = SCHOOL_FROST;
+    }
```

Comment:
`apply_affecting_aura` seems to be the new standard for rank 2 passives based on what I saw in other class files but that didn't seem to do the trick here.  The spell data for rank 2 seems to be more or less empty outside of the name so this made sense to me and I resorted to manually setting the school.

Reply:
You are setting the school to frost without setting it back to what it was before. And I am not sure how well supported adjusting school during combat is anyway.

It might also be just a different spell or additional damage as frost. But I am no DK and don't know how this exactly works in-game.

Reply:
Fixed it not setting back.  The way in game works right now is it just changes the school for the hit to frost.  Looking at logs, in game, and poking about spell data there's no separate spell involved for the frost damage other than [Killing Machine Rank 2](https://shadowlands.wowhead.com/spell=317214/killing-machine).  Its the largely the same SPELL_DAMAGE event in the raw log, just with a different school.  Confirmed that adjusting the school works by disabling all the other rank 2s and got the expected dps increase.

Reply:
This part is missing a check for Killing Machine's rank 2 on the player, and with this order of operation, only the obliterate after the one on which you spend KIlling Machine will be affected by the school change.
You should change the school to frost before the ``death_knight_melee_attack_t::execute();`` call, and set it back to physical after it.

Reply:
Fixed and confirmed in the debug log that the frost damage is on the right cast.  Not 100% sure I got the KM Rank 2 check right so that should be looked at.